src/engine/structures.rs

```diff
@@ -0,0 +1,156 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::collections::HashMap;
+use std::slice::IterMut;
+
+use uuid::Uuid;
+
+use super::engine::{HasName, HasUuid};
+
+
+/// Map UUID and name to T items.
+#[derive(Debug)]
+pub struct Table<T: HasName + HasUuid> {
+    items: Vec<T>,
+    name_map: HashMap<String, usize>,
+    uuid_map: HashMap<Uuid, usize>,
+}
```

Comment:
ok. btw I don't think I'd heard about the bug in bidirmap insert, can you elaborate?